### PR TITLE
Add pressure valve atmos device

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
New item the pressure valve
its a device that allow gas flow in a pipenet if the input pressure is higher than the set pressure

## Why It's Good For The Game
new atmos stuff is good

## Changelog
:cl:
add: Pressure Valve (a device that allow gas flow in a pipenet if the input pressure is higher than the set pressure)
/:cl:

---
Atomized from #53013

### CODE REVIEW

**Comment by SpaceManiac** [comment]
Please give your PRs descriptive titles.

**Comment by Ghilker** [comment]
Aww it had descriptive title

**Comment by halitosisman** [comment]
I wonder if this could finally make a filter-less SM setup possible.

**Comment by Ghilker** [comment]
In theory you could yes

**Review by Twaticus — APPROVED** [review]

**Comment by Maurukas** [comment]
I have been waiting for this for so long, and it will be helpful in building the engine the power design docs ask for.

**Comment by LemonInTheDark** [comment]
Looks alright to me, tested I trust?

**Comment by Ghilker** [comment]
tested extensively both on private and live server with the other PR Test merge, nothing seemed out of the ordinary, no runtime no weird stuff. Should be good to go

**Comment by Jared-Fogle** [comment]
I'm curious and don't do much atmos. What is this useful for?

**Comment by Ghilker** [comment]
This is like a pressure cooker valve, where when the pressure is high enough it releases gases. Its very useful in self controlled SM, it can be used as waste fire check. Also is a must fo the ultimate atmos autism the gas computer

**Comment by Dennok** [comment]
Great! I think about this thing and you code it! Thanks!

File: code/modules/atmospherics/machinery/components/binary_devices/pressure_valve.dm [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+/obj/machinery/atmospherics/components/binary/pressure_valve/process_atmos()
+
+	if(!on || !is_operational)
+		return
+
+	var/datum/gas_mixture/air1 = airs[1]
+	var/datum/gas_mixture/air2 = airs[2]
+
+	if(air1.return_pressure() > target_pressure)
+		if(air1.pump_gas_to(air2, air1.return_pressure()))
```
Comment (Dennok) [inline code/modules/atmospherics/machinery/components/binary_devices/pressure_valve.dm:60]: If we have pipe A and pipe B equal in volume, connected with this "pressure valve",
when pressure in A reach `target_pressure` it pump all gas from A to B ?

In my variant i think about just standard valve that opens self when pressure reached.
And closing self when pressure dropped.

File: code/modules/atmospherics/machinery/components/binary_devices/pressure_valve.dm [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+		if("pressure")
+			var/pressure = params["pressure"]
+			if(pressure == "max")
+				pressure = MAX_OUTPUT_PRESSURE
+				. = TRUE
+			else if(text2num(pressure) != null)
+				pressure = text2num(pressure)
+				. = TRUE
+			if(.)
+				target_pressure = clamp(pressure, 0, MAX_OUTPUT_PRESSURE)
```
Comment (Dennok) [inline code/modules/atmospherics/machinery/components/binary_devices/pressure_valve.dm:117]: I think MAX_OUTPUT_PRESSURE used for pump some small.

**Pull request merged**